src/refract/primitives.es6

```diff
@@ -0,0 +1,322 @@
+/*
+ * A refract element implementation with an extensible type registry.
```

Comment:
I think I'd prefer to default to not touching the value (i.e. assume it is already refracted). It's how the array type works today, and how I think the object type and other nested types should work. The contract is simple to understand and consistent: if you need to push or set a simple type then call `convertToType` first. If you need to set a more complex element, then use your element class and construct it yourself. I'd like to try that out and if it becomes a pain point then revisit this discussion.

Btw, I've pushed up support for attributes which are themselves elements (e.g. Resource.hrefVariables in the resource namespace). It allows custom types to register key names in the attributes hash that should be converted to types. Let me know what you think of it. There are some examples and a simple test in the latest commit.